addons/settings/fnc_parse.sqf

```diff
@@ -13,65 +13,58 @@ Returns:
     Settings with values and priority states. <ARRAY>
 
 Author:
-    commy2
+    commy2, johnb43
 ---------------------------------------------------------------------------- */
 
-// parses bool, number, string
-private _fnc_parseAny = {
-    params ["_string"];
-
-    // Remove whitespace so parseSimpleArray can handle arrays.
-    // Means that strings inside arrays don't support white space chars, but w/e.
-    // No such setting exists atm anyway.
-    if (_string find """" != 0) then {
-        _string = _string splitString WHITESPACE joinString "";
-    };
-
-    parseSimpleArray (["[", _string, "]"] joinString "") select 0
-};
-
 params [["_info", "", [""]], ["_validate", false, [false]], ["_source", "", [""]]];
 
-// remove whitespace at start and end of each line
-private _result = [];
+// Remove whitespace at start and end of each line, a line being define by the ";" at its end
+private _parsed = [];
 
 {
-    _result pushBack (_x call CBA_fnc_trim);
-} forEach (_info splitString NEWLINE);
+    _parsed pushBack (trim _x);
+} forEach (_info splitString ";");
 
-{
-    if (_x select [count _x - 1] != ";") then {
-        _result set [_forEachIndex, _x + ";"];
-    };
-} forEach _result;
+// Remove empty strings
+_parsed = _parsed - [""];
 
-_info = (_result joinString NEWLINE) + NEWLINE;
-
-// separate statements (setting = value)
-_result = [];
+// Separate statements (setting = value)
+private _result = [];
+private _indexEqualSign = -1;
+private _setting = "";
+private _priority = 0;
+private _value = "";
+private _countForce = count "force";
+private _whitespaces = [toString [ASCII_NEWLINE], toString [ASCII_CARRIAGE_RETURN], toString [ASCII_TAB], toString [ASCII_SPACE]];
```

Comment:
```suggestion
private _whitespaces = toString [ASCII_NEWLINE, ASCII_CARRIAGE_RETURN, ASCII_TAB, ASCII_SPACE];
```
`in` (L48 & 54) can search strings.

Reply:
Even better, there is a WHITE_SPACE macro already. 

https://github.com/CBATeam/CBA_A3/blob/master/addons/strings/script_strings.hpp#L13

So just _whitespace = toString WHITE_SPACE ;

Reply:
[WHITESPACE](https://github.com/CBATeam/CBA_A3/blob/c472830703a5c335bfa990c393a6b095d66d74bc/addons/settings/script_component.hpp#L149) already exists and is exactly what is wanted, so I used that instead.